lrlex/src/lib/ctbuilder.rs

```diff
@@ -1215,31 +1215,47 @@ pub fn ct_token_map<StorageT: Display + ToTokens>(
     let timestamp = env!("VERGEN_BUILD_TIMESTAMP");
     let mod_ident = format_ident!("{}", mod_name);
     write!(outs, "// lrlex build time: {}\n\n", quote!(#timestamp),).ok();
+    let storaget = str::parse::<TokenStream>(type_name::<StorageT>()).unwrap();
     // Sort the tokens so that they're always in the same order.
     // This will prevent unneeded rebuilds.
     let mut token_map_sorted = Vec::from_iter(token_map.borrow().iter());
     token_map_sorted.sort_by_key(|(k, _)| *k);
-    let tokens = &token_map_sorted
-        .into_iter()
+    let tokens = token_map_sorted
+        .iter()
         .map(|(k, id)| {
             let name = match rename_map {
                 Some(rmap) => *rmap.get(k.as_str()).unwrap_or(&k.as_str()),
-                _ => k,
+                _ => &k,
             };
             let tok_ident = format_ident!("T_{}", name.to_ascii_uppercase());
-            let storaget = str::parse::<TokenStream>(type_name::<StorageT>()).unwrap();
             // Code gen for the constant token values.
             quote! {
                 pub const #tok_ident: #storaget = #id;
             }
         })
         .collect::<Vec<_>>();
+    // Export an array of tokens to facilitate easier fuzzing.s
+    let token_array = &token_map_sorted
+        .iter()
+        .map(|(k, _)| {
+            let name = match rename_map {
+                Some(rmap) => *rmap.get(k.as_str()).unwrap_or(&k.as_str()),
+                _ => &k,
+            };
+            let tok_ident = format_ident!("T_{}", name.to_ascii_uppercase());
+            quote! {
+                #tok_ident,
+            }
+        })
```

Comment:
Since these are such similar cases, it might make sense to return a tuple

```
(quote!{pub const #tok_ident: #storaget = #id;}, quote!{#tok_ident})
```

then use [unzip](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.unzip) to make separate arrays